src/components/ExercisesEditPage.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component, PropTypes } from 'react';
+
+export default class ExercisesEditPage extends Component {
+
+  constructor() {
+    super();
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.fetchExercise(this.props.params.id);
+  }
+
+  componentWillUnmount() {
+    this.props.onPageLeave();
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    const { title, text, category, batch, fileUpload } = this.refs;
+    const { exercise, handleSubmit } = this.props;
+
+    handleSubmit(
+      exercise.get('id'),
+      title.value,
+      text.value,
+      category.value,
+      batch.value,
+      fileUpload.checked
+    );
+  }
+
+  renderOptionItems(set, chosenMember, refName) {
+    return (
+      <div className="field">
+        <select className="ui dropdown" ref={refName}>
+          <option value={set.get('id')}>
+              {chosenMember.get('title')}
+          </option>
+          {set
+            .valueSeq()
+            .filter(member => member.get('id') !== chosenMember.get('id'))
+            .map(member =>
+              <option key={member.get('id')} value={member.get('id')}>
+                {member.get('title')}
+              </option>
+            )
+          }
+        </select>
+      </div>
```

Comment:
That is a problem too. First, category is an optional. Second, we have an implicit default category/batch.
Empty option with empty value should be provided.